### DISCUSSION TITLE
htmx and syntax coloring editor

### DISCUSSION THREAD

**Discussion by mcesaro** [bigskysoftware/htmx#1652] [Q&A] [answered]
Hi there,
I'm looking for a way to implementa a syntax coloring editor based on [ace.js](https://ace.c9.io/).
My development setup is based on an erlang web server and template engine (similar to django).
I'm not able to find a way to have the content of a field into an instance of the Ace editor. What is not clear to me is the htmax approach to this.
I understand that with htmx is easy to quesy the server for the content to be edited, but I cannot edit with ace neither by defining the editor instance in the main html, nor in the rendered template.
The html code looks like this
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>aeon</title>
</head>
<script src="js/htmx.min.js"></script>
<script src="https://unpkg.com/hyperscript.org@0.9.7"></script>
<body>
    <div id="logheader" class="log-header">
        <div>
            <button hx-get="/v1/config" 
                    hx-trigger="click"
                    hx-target="#cfgEdit"
                    >
                    cfg
            </button>
        </div>
    </div>

    <div id="mainArea" style="display:inline-flex;">
        <div id="cfgEdit">
            <div name="editor" id="editor"> --- </div>
        </div>
    </div>

    <script src="js/ace.js" type="text/javascript" charset="utf-8">
        var editor = ace.edit("editor");
        editor.session.setMode("js/yaml");
    </script> 
</body>
</html>
```
and the template
```html
<div name="editor" id="editor"> --- </div>
<form action="/cfgsave" method="POST">
    <input type="submit" value="Test">
</form>
```
Now including a 
```html
<script>
    editor.setvalue({{cfgtext}});
</script>
```
in the template won't work as html rightly complains. Using hyperscript also seems not working but it's possible that I made some mistake on where and when to use it in the example.
I checked on SO but the more likely example was using htmx with djiango to integrate CKEditor in a modal dialog box and frankly I didn't undersant if that worked or not.
So I'm kindly asking the community if are there examples or references fo a similar integration.
Thanks

**Comment by matiboy** [comment, marked as answer]
Hi @mcesaro I did something similar recently though with Quill.js

I believe [this codepen does what you intend to](https://codepen.io/matiboy/pen/VwVqjgx). Basically, use `after-request` to put the content into your editor. You can do something similar for your "POST" to grab the content from the edit on `before-request`

Hope this helps

**Reply by mcesaro** [reply]
@matiboy  thank you so much.
I was able to apply your example to my situation and it worked like a charm!
Just for reference, as it might be of use to others, here is the full code of the sample:
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>aeon</title>
</head>
<script src="js/htmx.min.js"></script>

<script src="js/ace.js" type="text/javascript" charset="utf-8"></script>
<script>
    htmx.logAll();
    var aceEditor = null;
    document.addEventListener("DOMContentLoaded", function() {
        aceEditor = ace.edit(document.getElementById("editor"), {mode: "js/yaml"})
    });
</script>

<body>
    <div id="logheader" class="log-header">
        <div>
            <button hx-get="/v1/config" 
                    hx-on::after-request="aceEditor.setValue(event.detail.xhr.responseText)" 
                    hx-trigger="click" 
                    hx-swap="none"
                    name="config"
                    >
                    cfg
            </button>
        </div>
    </div>

    <div id="mainArea" style="display:inline-flex;">
        <div id="cfgEdit">
            <div id="editor"></div>
        </div>
    </div>

</body>

</html>
```
In my case what did the trick was to use `hx-swap="none"` to avoid replacing the button text with the response from the server.
